packages/@aws-cdk/aws-apigateway/lib/authorizers/identity-source.ts

```diff
@@ -0,0 +1,76 @@
+import { Token } from '@aws-cdk/core';
+
+/**
+ * Represents an identity source.
+ *
+ * The source can be specified either as a literal value (e.g: `Auth`) which
+ * cannot be blank, or as an unresolved string token.
+ */
+export class IdentitySource {
```

Comment:
We wouldn't be able to use tokens? If that's not a dealbreaker, then definitely I can make it simpler, but I did this deliberately with that in mind.

Reply:
Why do you think this will not work with tokens? 

I believe tokens in the cdk support string concatenation, such as `method.request.header.${Token[...]}`. Of course, the token must be a string token, and not of a different type, which I believe is the case here.

Let me know if I misunderstood.

Reply:
No I think that's right, not sure what I was thinking when I replied last night. Will go with the simpler version.